### PR TITLE
feat: diagnose

### PR DESCRIPTION
**Pull request by ammarkarachi**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/package.json [hunk 1]
```diff
@@ -40,6 +40,8 @@
     "@aws-amplify/amplify-util-uibuilder": "1.2.23",
     "@aws-amplify/graphql-auth-transformer": "^0.9.0",
     "@aws-cdk/cloudformation-diff": "~1.124.0",
+    "@types/archiver": "^5.3.1",
+    "@types/columnify": "^1.5.1",
```
Comment (jhockett) [inline packages/amplify-cli/package.json:44]: I think these should be in `devDependencies`

File: packages/amplify-cli/src/app-config/debug-config.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  private getCLIJson(throwIfNotExist: boolean = true): $TSAny {
+    const rootPath = pathManager.findProjectRoot();
+    if (!rootPath) {
+      if(!throwIfNotExist) {
+        return {}
+      }
+      throw new NotInitializedError();
+    }
+
+    const cliJson = stateManager.getCLIJSON(rootPath, undefined, { throwIfNotExist });
```
Comment (lazpavel) [inline packages/amplify-cli/src/app-config/debug-config.ts:42]: nit: `return stateManager.getCLIJSON(rootPath, undefined, { throwIfNotExist });`

File: .vscode/settings.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
   "jest.showCoverageOnLoad": true,
   "jest.runAllTestsFirst": false,
   "[typescript]": {
-    "editor.defaultFormatter": "dbaeumer.vscode-eslint"
```
[the remark this answers is not in the document]
Comment (ammarkarachi) [reply .vscode/settings.json:23]: This is not I might have done something wrong

File: packages/amplify-cli/src/commands/helpers/collect-files.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+const getCfnFiles = (resourceDir: string): string[] => {
+  const resourceBuildDir = path.join(resourceDir, 'build');
+
+  /**
+     * The API category w/ GraphQL builds into a build/ directory.
+     * This looks for a build directory and uses it if one exists.
+     * Otherwise falls back to the default behavior.
+     */
+  if (fs.existsSync(resourceBuildDir) && fs.lstatSync(resourceBuildDir).isDirectory()) {
```
Comment (sachscode) [inline packages/amplify-cli/src/commands/helpers/collect-files.ts:92]: Does this need to be here or should be a package level function ? 
installedPackages.map( package => package.collectFiles())

File: packages/amplify-cli/package.json [hunk 5]
```diff
@@ -82,6 +82,7 @@
     "env-editor": "^0.5.0",
     "execa": "^5.1.1",
     "folder-hash": "^4.0.2",
+    "form-data": "^4.0.0",
```
Comment (sachscode) [inline packages/amplify-cli/package.json:85]: Do we need to fix the package version here ?

File: packages/amplify-cli-logger/src/Redactor.ts [hunk 6]
```diff
@@ -27,7 +27,7 @@ export function Redactor(arg: string | undefined): string {
 
   return arg;
 }
-function stringMasker(s: string): string {
+export function stringMasker(s: string): string {
```
Comment (sachscode) [inline packages/amplify-cli-logger/src/Redactor.ts:30]: Can we add a function header which explains what this function is used for .

**Review by sachscode — COMMENTED** [review]
LGTM, just a few questions

File: packages/amplify-cli/src/commands/helpers/collect-files.ts (hunk 4 above)
Comment (sachscode) [reply packages/amplify-cli/src/commands/helpers/collect-files.ts:92]: Per discussion  this will be handled as part of context refactor.

**Pull request merged**